src/main/java/Logic.java

```diff
@@ -0,0 +1,61 @@
+import java.util.Scanner;
+
+public class Logic {
+    public static void loop() {
+        TaskList taskList = new TaskList();
+        while (true) {
+            Scanner scan = new Scanner(System.in);
+            String userInput = scan.nextLine();
+            Scanner secondScan = new Scanner(userInput);
+            String command = secondScan.next();
+            if (command.equals("bye")) {
+                break;
+            }
+            Output.printLine();
+            switch (command) {
+                case "list": {
+                    taskList.printTasks();
+                    break;
+                }
+                case "done": {
+                    int taskNumber = secondScan.nextInt();
+                    taskList.updateCompletion(taskNumber);
+                    break;
+                }
+                case "todo": {
+                    String taskName = secondScan.nextLine();
+                    Todo newTask = new Todo(taskName.trim());
+                    taskList.addTask(newTask);
+                    break;
+                }
+                case "deadline": {
```

Comment:
Extra indentation for case not required!

Reply:
Oh right, my bad on this. Will amend it asap :)